### ISSUE TITLE
Some doubts about the apply_rotary_emb function

### ISSUE THREAD

**Issue by zhuxiaopo** [GeeeekExplorer/nano-vllm#55]
`
def apply_rotary_emb(
    x: torch.Tensor,
    cos: torch.Tensor,
    sin: torch.Tensor,
) -> torch.Tensor:
    cos = cos.unsqueeze(-2)
    sin = sin.unsqueeze(-2)
    x1, x2 = torch.chunk(x.to(torch.float32), 2, dim=-1)
    y1 = x1 * cos - x2 * sin
    y2 = x2 * cos + x1 * sin
    return torch.cat((y1, y2), dim=-1).to(x.dtype)
`
The way this function is segmented seems strange，The original code uses torch.chunk (x, 2, dim=-1) to divide the input x into two parts (for example, 1024 dimensions are split into two 512 dimensions), but rope requires two adjacent dimensions to be a group (such as (D0, D1), (D2, D3),..., (D1022, D1023))

**Comment by cheunglei** [comment]
Thank you for your question.

Both approaches you mentioned for splitting the input tensor x are valid. The first method using `torch.chunk` (dividing into two equal parts) is the implementation used in Qwen3's RoPE implementation in the transformers library (reference: [Qwen3 implementation](https://github.com/huggingface/transformers/blob/v4.52.3/src/transformers/models/qwen3/modeling_qwen3.py#L94)). The second approach (pairing adjacent dimensions) is used in RoFormer's implementation (reference: [RoFormer implementation](https://github.com/huggingface/transformers/blob/v4.52.3/src/transformers/models/roformer/modeling_roformer.py#L300)).

To maintain consistency with Qwen3's implementation, we've adopted the first approach in our codebase.
